test/helpers/getPerformanceTests.ts

```diff
@@ -0,0 +1,79 @@
+import { evaluateXPathToNodes, evaluateXPathToString, evaluateXPathToFirstNode } from 'fontoxpath';
+
+import testFs from './testFs';
+import {
+	ALL_TESTS_QUERY,
+	getAllTestSets,
+	getFile,
+	unrunnableTestCasesByName,
+} from './qt3TestsTools';
+
+const PERFORMANCE_TESTS_FILE = 'runnablePerformanceTestNames.csv';
+const DEFAULT_NUMBER_OF_TEST = 50;
+
+const createNewRunnableTests = async (numberOfTest) => {
+	const allTestSets = getAllTestSets();
+	const allTestNames = [];
+	const newRunnableTests = new Set();
+
+	for (const testSetFileName of allTestSets) {
+		const testSet = getFile(testSetFileName);
+
+		// Find all the tests we can run
+		const testCases = evaluateXPathToNodes(ALL_TESTS_QUERY, testSet);
+
+		for (const testCase of testCases) {
+			try {
+				const testName = evaluateXPathToString('./@name', testCase);
+
+				if (unrunnableTestCasesByName[testName]) {
+					continue;
+				}
+
+				const assertionNode = evaluateXPathToFirstNode('./result/*', testCase);
+				const assertionType = assertionNode && (assertionNode as any).localName;
+
+				switch (assertionType) {
+					case 'assert':
+					case 'assert-true':
+					case 'assert-eq':
+					case 'assert-deep-eq':
+					case 'assert-empty':
+					case 'assert-false':
+					case 'assert-count':
+					case 'assert-type':
+					case 'assert-xml':
+					case 'assert-string-value':
+					case 'all-of':
+					case 'any-of':
+						allTestNames.push(testName);
+				}
+			} catch (e) {
+				console.error(e);
+				continue;
+			}
+		}
+	}
+
+	while (newRunnableTests.size < numberOfTest) {
+		const randomTestName = allTestNames[Math.floor(Math.random() * allTestNames.length)];
+		newRunnableTests.add(randomTestName);
+	}
+
+	testFs.writeFileSync(PERFORMANCE_TESTS_FILE, [...newRunnableTests].join('\n'));
+};
+
+export default function getPerformanceTests() {
+	if (process.argv.includes('--regenerate')) {
+		const newNumber = Number(process.argv[process.argv.length - 1]);
```

Comment:
let's use the parseInt function instead of the Number constructor. After that, you will only have to check for NaN.

Reply:
nice to know, thanks